### ISSUE TITLE
provide arm64 compiled version of `gen_snapshot_linux_x64_release` tool

### ISSUE THREAD

**Issue by moham96** [ardera/flutter-engine-binaries-for-arm#13]
Hello, 

Is it possible to provide an arm64 version of `gen_snapshot_linux_x64_release`? I have an arm64 laptop and want to generate the `app.so` on it instead of firing up my desktop PC.

Regards

**Comment by ardera** [comment]
Hey, theoretically that's possible but not sure I have a beefy enough ARM64 machine for that.
Github CI doesn't provide one at least and could be it takes ages to build on pi 4.

For now you'll have to use `qemu-user-static` I'm afraid.

**Comment by moham96** [comment]
> Hey, theoretically that's possible but not sure I have a beefy enough ARM64 machine for that. Github CI doesn't provide one at least and could be it takes ages to build on pi 4.
> 
> For now you'll have to use `qemu-user-static` I'm afraid.

Is the source code of the tool available ? so we can try to compile it ourselves

**Comment by ardera** [comment]
`gen_snapshot` is built as part of the flutter engine build.

For instructions see:
- https://github.com/flutter/flutter/wiki/Setting-up-the-Engine-development-environment
- https://github.com/flutter/flutter/wiki/Compiling-the-engine

**Comment by moham96** [comment]
> Hey, theoretically that's possible but not sure I have a beefy enough ARM64 machine for that. Github CI doesn't provide one at least and could be it takes ages to build on pi 4.
> 
> For now you'll have to use `qemu-user-static` I'm afraid.

on second thought, doesn't GitHub ci support cross-compiling?

**Comment by ardera** [comment]
of course, it's just a linux machine (you can do everything you can do on your home computer as well), but I _am_ already cross-compiling it: the `gen_snapshot` that generates snapshots for armv7 on a x64 linux machine is a side-product of cross-compiling the engine for armv7.

If you want to gen_snapshot itself to run on arm64 now you're basically cross-cross-compiling, which is not supported by the flutter build. Might be possible with some hacking but that'll probably break all the time.
